pkg/controller/rollout/rollout_canary.go

```diff
@@ -106,6 +166,12 @@ func (m *canaryReleaseManager) runCanary(c *RolloutContext) error {
 			return err
 		} else if done {
 			canaryStatus.CurrentStepState = v1beta1.CanaryStepStateTrafficRouting
+			// if it is partition style and the last batch, we can skip the CanaryStepStateTrafficRouting step
+			// to bypass the bug mentioned above
```

Comment:
plz be more specific about the bug, leave the bug url

Reply:
done